app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -113,30 +113,41 @@ class TourbookActivity : BaseNavigationActivity() {
     }
 
     private fun _import(uri: Uri) {
-        val confirmDialog = Dialog(this)
-        confirmDialog.setContentView(R.layout.dialog)
-        confirmDialog.findViewById<TextView>(R.id.dialogText).text = getString(
-                R.string.override_tourbook)
-        confirmDialog.findViewById<View>(R.id.yesButton).setOnClickListener {
-            try {
-                TourbookExporter(_db, contentResolver).importTourbook(uri)
-                Toast.makeText(this, getString(R.string.tourbook_import_successfull),
-                        Toast.LENGTH_SHORT).show()
-            } catch (e: Exception) {
-                Toast.makeText(this, getString(R.string.tourbook_import_error),
-                        Toast.LENGTH_SHORT).show()
+        val confirmDialog = AlertDialog.Builder(this).apply {
+            setTitle(R.string.warning)
+            setMessage(getString(R.string.override_tourbook))
+            setIcon(android.R.drawable.ic_dialog_alert)
+            setNegativeButton(R.string.cancel) { dialog, _ -> dialog.dismiss()}
+            setPositiveButton(R.string.ok) { dialog, _ ->
+                try {
+                    TourbookExporter(_db, contentResolver).importTourbook(uri)
+                    _makeToast(getString(R.string.tourbook_import_successfull))
+                } catch (e: Exception) {
+                    _showImportError(e.message.toString())
+                }
+                _initYears()
```

Comment:
Isn't there a `dialog.dismiss()` missing here?

Reply:
When the user touches any of the action buttons created with an AlertDialog.Builder, the system dismisses the dialog for you.